provider/lxd/storage.go

```diff
@@ -188,21 +189,41 @@ func (e *lxdStorageProvider) FilesystemSource(cfg *storage.Config) (storage.File
 }
 
 func ensureLXDStoragePool(env *environ, cfg *lxdStorageConfig) error {
-	createErr := env.raw.CreateStoragePool(cfg.lxdPool, cfg.driver, cfg.attrs)
-	if createErr == nil {
+	err := env.raw.CreateStoragePool(cfg.lxdPool, cfg.driver, cfg.attrs)
+	if err == nil {
 		return nil
-	}
-	// There's no specific error to check for, so we just assume
-	// that the error is due to the pool already existing, and
-	// verify that. If it doesn't exist, return the original
-	// CreateStoragePool error.
+	} else if ok := strings.Contains(err.Error(), "pool already exists"); !ok {
```

Comment:
I think it is safest to not rely on this string value as it may change under us.

Reply:
Maybe only option currently. Unless we can update the lxd rest api to return a more specific error code we may to settle for this hackish check for now.